### DISCUSSION TITLE
Key Remapping Issue for Firefox and Other Applications

### DISCUSSION THREAD

**Discussion by HiddenTT** [xremap/xremap#422] [Q&A]
Hello everyone! I'm having a problem with the configuration if someone could help me.

I am currently using Gnome on Debian12 with Wayland

Setting this configuration works for some applications and not for others, for example Firefox:

keymap:
   - name: Global Key Remapping
     remap:
       SUPER-c: CONTROL-c
       SUPER-v: CONTROL-v
       SUPER-a: CONTROL-a
      
If I establish the configuration for that specific application it still has no effect:

keymap:
   - name: Firefox Key Remapping
     application:
       only: Firefox-esr
     remap:
       SUPER-c: CONTROL-c
       SUPER-v: CONTROL-v
       SUPER-a: CONTROL-a
      
Could someone tell me what is happening?
Thank you so much!

**Comment by k0kubun** [comment]
Could you (re-)start xremap, press one of the remapped keys (e.g. SUPER-c), and report the whole output of the xremap command?

**Reply by HiddenTT** [reply]
It might be a problem related to events. If I use xev to capture the keyboard, it shows the following output:

![Captura desde 2024-02-26 11-27-09](https://github.com/k0kubun/xremap/assets/159443048/d008d477-39f1-467d-b5c2-57b1944cbbbe)

**Reply by k0kubun** [reply]
🤔 It's supposed to print out `application: ...` in the log when you press SUPER-c, but your output didn't.

The layer of `xev` is too high to debug xremap issues. Could you run `RUST_LOG=debug xremap ...` instead? That should print events in evdev, which xremap handles, instead of X11 events, which is generated after xremap emits keys.

**Comment by HiddenTT** [comment]
This is the result of pressing Super + C. However, the behavior is a bit strange, as, for example, it doesn't work in the search bar, but I can copy elements from the page. In Chrome, for example, it does work. In other applications, such as the text editor, it simply doesn't work :exploding_head: 

![WhatsApp Image 2024-02-26 at 23 05 07](https://github.com/k0kubun/xremap/assets/159443048/c2f7cb0f-d6fa-409e-8d13-b59991e44904)

**Reply by k0kubun** [reply]
Hmm, so it's recognized as Super-c, but xremap doesn't print `application:`. 

How did you install xremap? Is it `xremap-linux-x86_64-gnome.zip`, or `cargo install xremap --features gnome`?

**Comment by HiddenTT** [comment]
cargo install xremap --features gnome 

But if I run:
sudo /home/elcoco/Descargas/xremap-linux-x86_64-gnome/xremap /home/elcoco/.config/xremap/config.yml
It will have exactly the same effect

**Reply by k0kubun** [reply]
ok. Have you installed https://extensions.gnome.org/extension/5060/xremap/?

**Comment by HiddenTT** [comment]
Yes, of course :S

**Reply by k0kubun** [reply]
What's your GNOME version? Is the version part of the dropdown list when you installed the extension?

P.S. Could you reply to the same thread instead of making a new post (answer) for every reply? It's getting harder to follow the conversation.

**Reply by k0kubun** [reply]
I just got to read https://github.com/xremap/xremap-gnome/issues/6. The GNOME version might be the problem.

**Reply by k0kubun** [reply]
I just uploaded a new extension version that supports GNOME 46. It's being reviewed, but it might be available by the time when you read this. If you use GNOME 46, you might want to try that.

**Comment by alexindigo** [comment]
I'm experiencing similar issue on Niri, looks like applications filter doesn't work at all,

in my testing I map Super-C to Shift-Ctrl-C (or Ctrl_L-Insert), and it works if there is no applications filter, or if application filter is negative (e.g. `not: /ghostty/`) and it works in Ghostty too, if I put filter as `only: /ghostty/` – it doesn't work.

And I tried different ways to specify the app: 
- `[/ghostty/]`
- `["/ghostty/"]`
- `/ghostty/`
- `ghostty`
- `com.mitchellh.ghostty`

none of it works.
